test/data_type/test_custom_type.cpp

```diff
@@ -152,14 +208,42 @@ TEST(Custom_uint8, TestAsType)
     ck::static_for<0, size, 1>{}([&](auto i) {
         right_vec.template AsType<custom_uint8_t>()(Number<i>{}) = custom_uint8_t{test_vec.at(i)};
     });
-    // copy the vector
+    // copy the first half of a vector
```

Comment:
This change was rather unintended. You copy here whole vector. 
```suggestion
    // copy the vector
```